dev/src/reference.ts

```diff
@@ -2055,6 +2055,8 @@ export class CollectionReference extends Query {
     });
   }
 
+  doc(): DocumentReference;
+  doc(documentPath: string): DocumentReference;
```

Comment:
This weird arrangement here is so that our JSDoc generation still works. The TypeScript compiler still validates that we do not pass "undefined" (hence the test change).